src/App.js

```diff
@@ -1,16 +1,41 @@
 import React from 'react';
 import './App.css';
-import chatMessages from './data/messages.json';
+// import chatMessages from './data/messages.json';
+import chatMessages from './data/messages';
+import ChatLog from './components/ChatLog';
+
+
 
 const App = () => {
+  const [messages, setMessages] = React.useState(chatMessages)
+
+  const handleLike = (id) => {
+    setMessages(prevMessages => {
+      const updatedMessages = prevMessages.map(message => {
+        return message.id === id ? {...message, liked: !message.liked} : message
+      })
+      return updatedMessages
+    })
+  }
+
+  const totalLikes = () => {
+    let total = 0;
+    for (let message of messages) {
+      if (message.liked) {total += 1}
+    }
+
+    return total;
+  };
+
+
   return (
     <div id="App">
       <header>
-        <h1>Application title</h1>
+        <h1>Messages to Vladimir</h1>
       </header>
       <main>
-        {/* Wave 01: Render one ChatEntry component
-        Wave 02: Render ChatLog component */}
+        <h3>Total Likes: {totalLikes()}  ❤️s </h3>
+        <ChatLog entries={messages} handleLike={handleLike}/>
       </main>
     </div>
```

Comment:
Great job everyone! You both did well on this project! There wasn't much to correct on this one! Let me know if you need to discuss any of my comments in further detail! Congrats! ✨